### PR TITLE
Add "new_name" parameter to repo transfer request (#2787)

### PR DESCRIPTION
**Pull request by kgalli**
Resolves: https://github.com/google/go-github/issues/2787

### CODE REVIEW

File: github/repos.go [hunk 1]
```diff
@@ -1969,6 +1969,7 @@ func (s *RepositoriesService) RemoveUserRestrictions(ctx context.Context, owner,
 // TransferRequest represents a request to transfer a repository.
 type TransferRequest struct {
 	NewOwner string  `json:"new_owner"`
+	NewName  string  `json:"new_name,omitempty"`
```
Comment (gmlewis) [inline github/repos.go:1972]:
```suggestion
	NewName  *string  `json:"new_name,omitempty"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @kgalli !
After you make the requested change, please run "go generate ./..." and push (not force-push) the results to this PR, as described in CONTRIBUTING.md.

**Comment by kgalli** [comment]
@gmlewis thanks for the hint. I updated the code and run the generate command. Have to admit did not thought about the accessors. Very nice that you have generators in place for that.

**Review by gmlewis — APPROVED** [review]
Thank you, @kgalli !
LGTM.
Merging after tests pass.

**Pull request merged**